notebooks/ReFT_Adapters_Finetuning.ipynb

```diff
@@ -0,0 +1,505 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datasets import load_dataset\n",
+    "\n",
+    "ds = load_dataset(\"nyu-mll/glue\", \"mnli\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dict_keys(['train', 'validation_matched', 'validation_mismatched', 'test_matched', 'test_mismatched'])\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(ds.keys())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will use the `train` and `validation_matched` splits for our training and testing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_dataset = ds[\"train\"]\n",
+    "eval_dataset = ds[\"validation_matched\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#initialize the model\n",
+    "model_name_or_path = \"roberta-base\"\n",
+    "from transformers import AutoTokenizer\n",
+    "tokenizer = AutoTokenizer.from_pretrained(model_name_or_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Data Preprocessing\n",
+    "\n",
+    "We will return the example and the hypothesis statements with max padding, and remove the column 'idx' as it is not needed for training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def preprocess_function(example):\n",
+    "    return tokenizer(example['premise'], example['hypothesis'], return_tensors = \"pt\", truncation=True, padding='max_length')\n",
+    "\n",
+    "train_dataset = train_dataset.map(preprocess_function, batched=True)\n",
+    "eval_dataset = eval_dataset.map(preprocess_function, batched=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_dataset = train_dataset.remove_columns([\"idx\"])\n",
+    "eval_dataset = eval_dataset.remove_columns([\"idx\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import default_data_collator\n",
+    "\n",
+    "data_collator = default_data_collator"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Model and Adapter initialization\n",
+    "\n",
+    "We load the `roberta-base` model along with the `LoReftConfig`. We can initalize a `reft` config with only one line of code, and can add it to our base model using the `add_adapter` function. On top of that, we can add a classification head to our adapter specifying 3 labels.\n",
```

Comment:
could be nice to link our docs on ReFT here: https://docs.adapterhub.ml/methods.html#reft and mention that there is explanation on supported config parameters.